### PR TITLE
Use java 17 during %check build stage.

### PR DESCRIPTION
**Pull request by toregge**
@aressem : please review

### CODE REVIEW

File: dist/vespa.spec [hunk 1]
```diff
@@ -466,6 +466,15 @@ VERSION=%{version} CI=true make -C client/go install-all
...
 %check
 %if ! 0%{?installdir:1}
+%if 0%{?_java_home:1}
+export JAVA_HOME=%{?_java_home}
+%else
+export JAVA_HOME=/usr/lib/jvm/java-17-openjdk
+%endif
+export PATH="$JAVA_HOME/bin:$PATH"
+%if 0%{?el8}
+python3.9 -m pip install --user pytest
```
Comment (aressem) [inline dist/vespa.spec:476]: Not a good idea to install stuff as part of the rpmbuild ?

**Review by aressem — CHANGES_REQUESTED** [review]

File: dist/vespa.spec (hunk 1 above)
Comment (toregge) [reply dist/vespa.spec:476]: Not installing pytest causes a failure due to pytest not being installed.

Comment (aressem) [reply dist/vespa.spec:476]: Ok. I do wish we could express this dependency with `BuildRequires`.

**Review by aressem — APPROVED** [review]

File: dist/vespa.spec (hunk 1 above)
Comment (baldersheim) [reply dist/vespa.spec:476]: Is this due to ANN testing ?

Comment (aressem) [reply dist/vespa.spec:476]: It is for the ANN benchmark unit test.

**Pull request merged**